collector/generic_collector.go

```diff
@@ -78,6 +78,10 @@ func NewCollector(collectorName string, configfile string) (*GenericCollector, e
 			return nil, fmt.Errorf("Invalid regexp %v for metric %v", metricConfig.Regex, metricConfig.Name)
 		}
 	}
+	//set minPollFrequency to maxHousekeepingInterval if config returns minpollFrequency=0
+	if minPollFrequency == 0 {
```

Comment:
I think we should move this logic up to replace line 68. Start with the default and then compare in the loop (#72) and lower it if we find a user-defined value.